contracts/libraries/Fixed.sol

```diff
@@ -532,6 +532,73 @@ library FixLib {
             return uint192(shiftDelta / FIX_ONE); // {D18} = {D36} / {D18}
         }
     }
+
+    /// Divide two fixes, rounding up to FIX_MAX and down to 0
+    /// @param a Numerator
+    /// @param b Denominator
+    function safeDiv(
+        uint192 a,
+        uint192 b,
+        RoundingMode rounding
+    ) internal pure returns (uint192) {
+        if (a == 0) return 0;
+        if (b == 0) return FIX_MAX;
+
+        uint256 raw = _divrnd(FIX_ONE_256 * a, uint256(b), rounding);
+        if (raw >= FIX_MAX) return FIX_MAX;
+        return uint192(raw); // don't need _safeWrap
+    }
+
+    /// Multiplies two fixes and divide by a third
+    /// @param a First to multiply
+    /// @param b Second to multiply
+    /// @param c Denominator
+    function safeMulDiv(
+        uint192 a,
+        uint192 b,
+        uint192 c,
+        RoundingMode rounding
+    ) internal pure returns (uint192 result) {
+        if (a == 0 || b == 0) return 0;
+        if (a == FIX_MAX || b == FIX_MAX || c == 0) return FIX_MAX;
+
+        uint256 result_256;
+        unchecked {
+            (uint256 hi, uint256 lo) = fullMul(a, b);
+            if (hi >= c) return FIX_MAX;
```

Comment:
this line and L595 are the only 2 that feel a bit odd to me, but we already follow a similar pattern in `safeMul`, so i think this is correct to return FIX_MAX instead of erroring.